src/printer.js

```diff
@@ -1442,8 +1442,11 @@ function genericPrintNoParens(path, options, print) {
       ]);
     case "DeclareInterface":
     case "InterfaceDeclaration": {
-      const parent = path.getParentNode(1);
-      if (parent && parent.type === "DeclareModule") {
+      const nodeSource = options.originalText.slice(0, util.locStart(n));
+      if (
+        n.type === "DeclareInterface" ||
+          (options.parser === "flow" && nodeSource.match(/declare\s*$/))
```

Comment:
Okay, last small thing and should be good to go, would be nice to put the flow detection code a function, so we don't have to copy and paste this crazy logic twice.

Also, it means we don't do the slice in case we are in babylon.